### ISSUE TITLE
[ads] Search result ads are not landed

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#38318]
Important: This a regression. Not reproducible in 1.65.x.

## Steps to Reproduce

1. Clean profile
1. Trigger a search ad
1. Click the ad
1. Wait till it is landed

## Actual result:

`Maybe land` is not triggered

## Expected result:
`Maybe land` is triggered

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

Brave | 1.66.107 Chromium: 125.0.6422.41 (Official Build) (64-bit)
-- | --
Revision | 188213f024297511fc5e37efe56eabde5992b300
OS | Linux

cc @tmancey @aseren @iambrianfung

**Comment by kjozwiak** [comment]
The above requires `1.66.10` or higher for `1.66.x` verification 👍

**Comment by btlechowski** [comment]
Verified with

Brave | 1.66.110 Chromium: 125.0.6422.60 (Official Build) (64-bit)
-- | --
Revision | de593d76e2ca4d02faa85bf1ca27bcf3ee46793c
OS | Linux

Verified test plan from https://github.com/brave/brave-browser/issues/38318#issue-2295805234


#### non-rewards users

<details>

```
[6955:6955:0516/121156.950518:VERBOSE1:redeem_non_reward_confirmation.cc(57)] Redeem non-reward confirmation
[6955:6955:0516/121156.950787:VERBOSE1:redeem_non_reward_confirmation.cc(66)] Create non-reward confirmation
[6955:6955:0516/121156.950950:VERBOSE6:redeem_non_reward_confirmation.cc(71)] URL Request:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7
  Content: {"creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a","transactionId":"1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7","type":"landed"}
  Content Type: application/json
  Method: kPost
[6955:6955:0516/121156.951473:VERBOSE7:redeem_non_reward_confirmation.cc(72)]   Headers:
    accept: application/json
[6955:6955:0516/121157.193398:VERBOSE6:redeem_non_reward_confirmation.cc(85)] URL Response:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7
  Response Status Code: 418
  Response: {"id":"1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7","payload":{},"type":"landed","createdAt":"2024-05-16T10:11:56.902Z","modifiedAt":"2024-05-16T10:11:56.902Z","creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a"}
[6955:6955:0516/121157.193626:VERBOSE7:redeem_non_reward_confirmation.cc(86)]   Headers:
    accept-ranges: bytes
    access-control-allow-credentials: true
    content-length: 213
    content-security-policy: sandbox
    content-type: application/json; charset=utf-8
    cross-origin-opener-policy: same-origin
    cross-origin-resource-policy: same-origin
    date: Thu, 16 May 2024 10:11:57 GMT
    referrer-policy: same-origin
    strict-transport-security: max-age=31536000; includeSubDomains; preload
    vary: Origin
    via: 1.1 varnish
    x-cache: MISS
    x-cache-hits: 0
    x-content-type-options: nosniff
    x-frame-options: DENY
    x-served-by: cache-hel1410028-HEL
[6955:6955:0516/121157.193742:VERBOSE1:redeem_non_reward_confirmation.cc(106)] Successfully redeemed non-reward landed confirmation for search_result_ad with transaction id 1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7 and creative instance id bc876b1a-ca48-4eac-b445-efb9617ead4a
[6955:6955:0516/121157.204963:VERBOSE1:confirmations.cc(97)] Successfully processed landed confirmation for search_result_ad with transaction id 1a2caf4c-51a3-4ba3-83a3-60b4b21c34a7 and creative instance id bc876b1a-ca48-4eac-b445-efb9617ead4a

```


</details>


#### rewards users

<details>

```
[6955:6955:0516/121747.105930:VERBOSE1:redeem_reward_confirmation.cc(71)] Redeem reward confirmation
[6955:6955:0516/121747.108832:VERBOSE1:redeem_reward_confirmation.cc(86)] Create reward confirmation
[6955:6955:0516/121747.109079:VERBOSE6:redeem_reward_confirmation.cc(91)] URL Request:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb/eyJzaWduYXR1cmUiOiJybHd3Wms4SkxRaS9IYm11RVlRZTRuWXRtTXZLUUM1REZNMHh4MUZsaDdLOEVOdXNEN1N0TFUxMEswYkV2R3BpT0ZaUmxFY1pDejNQRnlwdkNSRXA3Zz09IiwidCI6IlpwUE96NUxNNHRmNS9YNUZYbXdvQmJOaWZIaW9teUJFMGp1UUVmT0xHTzQ4NjMyZ3dheHJ5YXVrbFJORW1CVTRZUWR0QzJNQXFvbk5VMVFZOUxWVkJBPT0ifQ==
  Content: {"blindedPaymentTokens":["7hjx09EqnP9I6jB2p/6k9FLs5pLIrWmxL8NvrYDOhTI="],"buildChannel":"release","catalog":[{"id":"iJLmlBgU36fPLgTp+S7Rl+dPcIio4iV1I8gIj3sCV48="}],"countryCode":"US","createdAtTimestamp":"2024-05-16T10:00:00.000Z","creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a","platform":"linux","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotating_hash":"/zJrAMhGmp8nWfVOkbc6Hlh9XommCqGaFMIlQmwUDKM=","segment":"keyword","studies":[],"systemTimestamp":"2024-05-16T10:00:00.000Z","transactionId":"4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb","type":"landed","versionNumber":"125.0.6422.60"}
  Content Type: application/json
  Method: kPost
[6955:6955:0516/121747.109339:VERBOSE7:redeem_reward_confirmation.cc(92)]   Headers:
    accept: application/json
[6955:6955:0516/121747.357688:VERBOSE6:redeem_reward_confirmation.cc(105)] URL Response:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb/eyJzaWduYXR1cmUiOiJybHd3Wms4SkxRaS9IYm11RVlRZTRuWXRtTXZLUUM1REZNMHh4MUZsaDdLOEVOdXNEN1N0TFUxMEswYkV2R3BpT0ZaUmxFY1pDejNQRnlwdkNSRXA3Zz09IiwidCI6IlpwUE96NUxNNHRmNS9YNUZYbXdvQmJOaWZIaW9teUJFMGp1UUVmT0xHTzQ4NjMyZ3dheHJ5YXVrbFJORW1CVTRZUWR0QzJNQXFvbk5VMVFZOUxWVkJBPT0ifQ==
  Response Status Code: 201
  Response: {"id":"4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb","payload":{},"type":"landed","createdAt":"2024-05-16T10:17:47.063Z","modifiedAt":"2024-05-16T10:17:47.063Z","creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a"}
[6955:6955:0516/121747.358080:VERBOSE7:redeem_reward_confirmation.cc(106)]   Headers:
    accept-ranges: bytes
    access-control-allow-credentials: true
    content-length: 213
    content-security-policy: sandbox
    content-type: application/json; charset=utf-8
    cross-origin-opener-policy: same-origin
    cross-origin-resource-policy: same-origin
    date: Thu, 16 May 2024 10:17:47 GMT
    referrer-policy: same-origin
    strict-transport-security: max-age=31536000; includeSubDomains; preload
    vary: Origin
    via: 1.1 varnish
    x-cache: MISS
    x-cache-hits: 0
    x-content-type-options: nosniff
    x-frame-options: DENY
    x-served-by: cache-hel1410028-HEL
[6955:6955:0516/121747.358198:VERBOSE1:redeem_reward_confirmation.cc(115)] Fetch payment token
[6955:6955:0516/121747.358444:VERBOSE6:redeem_reward_confirmation.cc(119)] URL Request:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb/paymentToken
  Method: kGet
[6955:6955:0516/121747.358797:VERBOSE7:redeem_reward_confirmation.cc(120)]   Headers:

[6955:6955:0516/121747.577174:VERBOSE6:redeem_reward_confirmation.cc(133)] URL Response:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb/paymentToken
  Response Status Code: 200
  Response: {"id":"4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb","type":"landed","createdAt":"2024-05-16T10:17:24.925Z","modifiedAt":"2024-05-16T10:17:27.230Z","creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a","paymentToken":{"publicKey":"vlGTZ+NzScItHa2Z4YszNVzvOC11WYmiQ1zLZ84cl20=","batchProof":"jLn7ENCLnEFP389+rVjzEDN9EX8w8Vu3LII6OEWoywo++nHzHC4k8+6MZV1LTU2QIHeFvhVj2zqbY/vlhqv1Cg==","signedTokens":["KGsBPy2DN2m/pIt7G/mnjdcuPB1knOqhVuk9YYrWOjs="]}}
[6955:6955:0516/121747.577627:VERBOSE7:redeem_reward_confirmation.cc(134)]   Headers:
    accept-ranges: bytes
    access-control-allow-credentials: true
    content-length: 444
    content-security-policy: sandbox
    content-type: application/json; charset=utf-8
    cross-origin-opener-policy: same-origin
    cross-origin-resource-policy: same-origin
    date: Thu, 16 May 2024 10:17:47 GMT
    referrer-policy: same-origin
    strict-transport-security: max-age=31536000; includeSubDomains; preload
    vary: Origin, Accept-Encoding
    via: 1.1 varnish
    x-cache: MISS
    x-cache-hits: 0
    x-content-type-options: nosniff
    x-frame-options: DENY
    x-served-by: cache-hel1410028-HEL
[6955:6955:0516/121747.580399:VERBOSE9:confirmation_state_manager.cc(79)] Saving confirmation state
[6955:6955:0516/121747.582741:VERBOSE1:redeem_reward_confirmation_util.cc(48)] You have 3 payment tokens which will be redeemed in 0 hours, 1 minute, 0 seconds at 12:18:47.401
[6955:6955:0516/121747.587537:VERBOSE9:confirmation_state_manager.cc(87)] Successfully saved confirmation state
[6955:6955:0516/121747.600047:VERBOSE1:confirmations.cc(97)] Successfully processed landed confirmation for search_result_ad with transaction id 4b34d04b-0f3e-4f8e-a5df-a85e6538b4bb and creative instance id bc876b1a-ca48-4eac-b445-efb9617ead4a


```


</details>

**Comment by btlechowski** [comment]
Verified on 1.68.18 on Lenovo TB-8506FS (Android 11.0)

#### non-rewards users

<details>

```



12:39:08.940  V  [VERBOSE1:redeem_non_reward_confirmation.cc(57)] Redeem non-reward confirmation
12:39:08.940  V  [VERBOSE1:redeem_non_reward_confirmation.cc(66)] Create non-reward confirmation
12:39:08.941  V  [VERBOSE6:redeem_non_reward_confirmation.cc(71)] URL Request:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/9a930406-7120-444b-956b-3a86cd8cbb56
                   Content: {"creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a","transactionId":"9a930406-7120-444b-956b-3a86cd8cbb56","type":"landed"}
                   Content Type: application/json
                   Method: kPost
12:39:09.862  V  [VERBOSE6:redeem_non_reward_confirmation.cc(85)] URL Response:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/9a930406-7120-444b-956b-3a86cd8cbb56
                   Response Status Code: 418
                   Response: {"id":"9a930406-7120-444b-956b-3a86cd8cbb56","payload":{},"type":"landed","createdAt":"2024-05-16T10:39:09.887Z","modifiedAt":"2024-05-16T10:39:09.887Z","creativeInstanceId":"bc876b1a-ca48-4eac-b445-efb9617ead4a"}
12:39:09.863  V  [VERBOSE1:redeem_non_reward_confirmation.cc(106)] Successfully redeemed non-reward landed confirmation for search_result_ad with transaction id 9a930406-7120-444b-956b-3a86cd8cbb56 and creative instance id bc876b1a-ca48-4eac-b445-efb9617ead4a
12:39:09.884  V  [VERBOSE1:confirmations.cc(97)] Successfully processed landed confirmation for search_result_ad with transaction id 9a930406-7120-444b-956b-3a86cd8cbb56 and creative instance id bc876b1a-ca48-4eac-b445-efb9617ead4a

```

</details>



#### rewards users

<details>


```

12:44:59.905  V  [VERBOSE1:redeem_reward_confirmation.cc(71)] Redeem reward confirmation
12:44:59.915  V  [VERBOSE1:redeem_reward_confirmation.cc(86)] Create reward confirmation
12:44:59.916  V  [VERBOSE6:redeem_reward_confirmation.cc(91)] URL Request:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/f996e53a-074e-46bd-8078-75ef6d6362e1/eyJzaWduYXR1cmUiOiIvMWc3OWxqWTZTR0xUS3NUVEJSVGROTWhjYnF1RGFIQ1FETW9XY0prS3phWE9RR0tzRDZuMlI4K0tTNWQ4RWJPTWR4M0dldEYzN1NpU0pLVFhyZk9pQT09IiwidCI6Ims3VGJuZjFTaklUNGhBMWY4cVRHSlE5U3E4WTRxUmtqV1BWWTUwaVcwQnNwRU1QcjVpVWMzTTFTdDNlY1c0aFpEa0ppZkVwRVRFUGxDQ1FhZEZhYWJRPT0ifQ==
                   Content: {"blindedPaymentTokens":["HPsuSCCqUlpYdk5GA/gUQ1zLSSjhpbTZKE8JERtHGRg="],"buildChannel":"release","catalog":[{"id":"iJLmlBgU36fPLgTp+S7Rl+dPcIio4iV1I8gIj3sCV48="}],"countryCode":"PL","createdAtTimestamp":"2024-05-16T10:00:00.000Z","creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","platform":"android","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotating_hash":"x6iEjfvDMv6noTGvtAy1yAeMPtZTXh+iGQ6O3jtBAvo=","segment":"keyword","studies":[],"systemTimestamp":"2024-05-16T10:00:00.000Z","transactionId":"f996e53a-074e-46bd-8078-75ef6d6362e1","type":"landed","versionNumber":"125.0.6422.60"}
                   Content Type: application/json
                   Method: kPost
12:45:00.280  V  [VERBOSE6:redeem_reward_confirmation.cc(105)] URL Response:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/f996e53a-074e-46bd-8078-75ef6d6362e1/eyJzaWduYXR1cmUiOiIvMWc3OWxqWTZTR0xUS3NUVEJSVGROTWhjYnF1RGFIQ1FETW9XY0prS3phWE9RR0tzRDZuMlI4K0tTNWQ4RWJPTWR4M0dldEYzN1NpU0pLVFhyZk9pQT09IiwidCI6Ims3VGJuZjFTaklUNGhBMWY4cVRHSlE5U3E4WTRxUmtqV1BWWTUwaVcwQnNwRU1QcjVpVWMzTTFTdDNlY1c0aFpEa0ppZkVwRVRFUGxDQ1FhZEZhYWJRPT0ifQ==
                   Response Status Code: 201
                   Response: {"id":"f996e53a-074e-46bd-8078-75ef6d6362e1","payload":{},"type":"landed","createdAt":"2024-05-16T10:45:00.276Z","modifiedAt":"2024-05-16T10:45:00.276Z","creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8"}
12:45:00.282  V  [VERBOSE1:redeem_reward_confirmation.cc(115)] Fetch payment token
12:45:00.282  V  [VERBOSE6:redeem_reward_confirmation.cc(119)] URL Request:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/f996e53a-074e-46bd-8078-75ef6d6362e1/paymentToken
                   Method: kGet
12:45:00.529  V  [VERBOSE6:redeem_reward_confirmation.cc(133)] URL Response:
                   URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/f996e53a-074e-46bd-8078-75ef6d6362e1/paymentToken
                   Response Status Code: 200
                   Response: {"id":"f996e53a-074e-46bd-8078-75ef6d6362e1","type":"landed","createdAt":"2024-05-16T10:44:41.765Z","modifiedAt":"2024-05-16T10:44:43.963Z","creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","paymentToken":{"publicKey":"vlGTZ+NzScItHa2Z4YszNVzvOC11WYmiQ1zLZ84cl20=","batchProof":"gMO9+h9K64SBANk6DqulaIfrzuwCaVfbQm1mQmRXvwty3iLCNP38egtKYaAJEP0bXDrhT7OG+htJ6+KtjnSoBQ==","signedTokens":["4j8nZ7wmwNxmdxAhJa/ekPsRYqN7bW6e+oHSpjunp04="]}}
12:45:00.558  V  [VERBOSE1:redeem_reward_confirmation_util.cc(48)] You have 3 payment tokens which will be redeemed in 0 godzin, 0 minut i 24 sekundy at 12:45:24,442
12:45:00.570  V  [VERBOSE1:confirmations.cc(97)] Successfully processed landed confirmation for search_result_ad with transaction id f996e53a-074e-46bd-8078-75ef6d6362e1 and creative instance id 788a9319-9ff9-4cf8-944b-98dcdc3dbfa8

```


</details>
